code/datums/components/crafting/_recipes.dm

```diff
@@ -0,0 +1,40 @@
+
+/datum/crafting_recipe
+	///in-game display name -  Optional, if not set uses result name
+	var/name = ""
+	///type paths of items consumed associated with how many are needed
+	var/reqs[] = list()
+	//type paths of items explicitly not allowed as an ingredient
+	var/blacklist[] = list()
+	//type path of item resulting from this craft
+	var/result
+	//type paths of items needed but not consumed
+	var/tools[] = list()
+	//time in deciseconds
+	var/time = 30
```

Comment:
If you are moving time around, you might as well change this to 3 SECONDS and use time defines instead.